packages/gatsby/src/redux/actions.js

```diff
@@ -1134,15 +1134,22 @@ actions.createRedirect = ({
 
   // Parse urls to get their protocols
   // url.parse will not cover protocol-relative urls so do a separate check for those
-  const parsed = url.parse(toPath)
-  const isRelativeProtocol = toPath.startsWith(`//`)
+  const parsedFromPath = url.parse(fromPath)
+  const isFromPathRelativeProtocol = fromPath.startsWith(`//`)
+  const fromPathPrefix =
+    parsedFromPath.protocol != null || isFromPathRelativeProtocol
+      ? ``
+      : pathPrefix
+
+  const parsedToPath = url.parse(toPath)
+  const isToPathRelativeProtocol = toPath.startsWith(`//`)
   const toPathPrefix =
-    parsed.protocol != null || isRelativeProtocol ? `` : pathPrefix
+    parsedToPath.protocol != null || isToPathRelativeProtocol ? `` : pathPrefix
```

Comment:
Can we abstract path normalization to function that would be used by both `fromPath` and `toPath` to not repeat same code twice?

something like:
```js
/**
 * Check if path is absolute and add pathPrefix in front if it's not
 */
const maybeAddPathPrefix = (path, pathPrefix) => {
  const parsedToPath = url.parse(path)
  const isPathRelativeProtocol = path.startsWith(`//`)
  return `${parsedToPath.protocol != null || isPathRelativeProtocol ? `` : pathPrefix}${path}`
}
```
And then 

```js
  return {
    type: `CREATE_REDIRECT`,
    payload: {
      fromPath: maybeAddPathPrefix(fromPath, pathPrefix),
      isPermanent,
      redirectInBrowser,
      toPath: maybeAddPathPrefix(toPath, pathPrefix),
      ...rest,
    },
  }
```

Reply:
Thanks for the review. I updated the code.